migrate/src/main/scala/migrate/internal/MigrationFile.scala

```diff
@@ -0,0 +1,62 @@
+package migrate.internal
+
+import interfaces.CompilationUnit
+import scalafix.interfaces._
+import migrate.utils.FileUtils
+import scala.util.{Try, Failure, Success}
+import migrate.utils.ScalaExtensions._
+import migrate.AbsolutePath
+
+private[migrate] trait MigrationFile {
+  def source: AbsolutePath
+  def patches: Seq[ScalafixPatch]
+  def previewAllPatches(): Try[CompilationUnit]
+  def previewPatches(patches: Seq[ScalafixPatch]): Try[CompilationUnit]
+}
+
+private[migrate] object MigrationFile {
+  def apply(source: AbsolutePath, evaluation: Option[ScalafixFileEvaluation]): MigrationFile = {
+    evaluation
+      .map(ScalafixMigrationFile(source, _))
+      .getOrElse(DefaultMigrationFile(source))
+  }
+}
+
+private[migrate] case class ScalafixMigrationFile(source: AbsolutePath, evaluation: ScalafixFileEvaluation) extends MigrationFile {
+  val patches: Seq[ScalafixPatch] = evaluation.getPatches().toSeq
+
+  def previewAllPatches(): Try[CompilationUnit] = {
+    evaluation.previewPatches()
+      .map { content => 
+        new CompilationUnit(source.value, content)
+      }
+      .asScala
+      .map(Success(_))
+      .getOrElse {
+        Failure(new ScalafixException(s"Cannot apply patch on file $source"))
+      }
+  }
+
+  def previewPatches(patches: Seq[ScalafixPatch]): Try[CompilationUnit] = {
+    evaluation.previewPatches(patches.toArray)
+      .map { content => 
+        new CompilationUnit(source.value, content)
+      }
+      .asScala
+      .map(Success(_))
```

Comment:
let add a toTry extension method 
```
    def toTry(e: => Throwable): Try[A] = in match {
      case Some(v) => Success(v)
      case None => Failure(e)
    }
```
to replace 
.asScala
.toTry(new ScalafixException(s"Cannot apply patch on file $source"))

Reply:
:100: